scripts/installer/curl_install_pypi/install.py

```diff
@@ -0,0 +1,360 @@
+    print_status('Python version {}.{}.{} okay.'.format(v.major, v.minor, v.micro))
+
+
+def _native_dependencies_for_dist(verify_cmd_args, install_cmd_args, dep_list):
+    try:
+        print_status("Executing: '{} {}'".format(' '.join(verify_cmd_args), ' '.join(dep_list)))
+        subprocess.check_output(verify_cmd_args + dep_list, stderr=subprocess.STDOUT)
+        print_status('Native dependencies okay.')
+    except subprocess.CalledProcessError:
+        err_msg = 'One or more of the following native dependencies are not currently installed and may be required.\n'
+        err_msg += '"{}"'.format(' '.join(install_cmd_args + dep_list))
+        print_status(err_msg)
+        ans_yes = prompt_y_n('Missing native dependencies. Attempt to continue anyway?', 'n')
+        if not ans_yes:
+            raise CLIInstallError('Please install the native dependencies and try again.')
+
+
+def _get_linux_distro():
+    if platform.system() != 'Linux':
+        return None, None
+
+    try:
+        with open('/etc/os-release') as lines:
+            tokens = [line.strip() for line in lines]
+    except Exception:
+        return None, None
+
+    release_info = {}
+    for token in tokens:
+        if '=' in token:
+            k, v = token.split('=', 1)
+            release_info[k.lower()] = v.strip('"')
+
+    return release_info.get('name', None), release_info.get('version_id', None)
+
+
+def verify_install_dir_exec_path_conflict(install_dir, exec_dir):
+    for exec_name in [PF_EXECUTABLE_NAME, PFAZURE_EXECUTABLE_NAME, PFS_EXECUTABLE_NAME]:
+        exec_path = os.path.join(exec_dir, exec_name)
+        if install_dir == exec_path:
+            raise CLIInstallError("The executable file '{}' would clash with the install directory of '{}'. Choose "
+                                  "either a different install directory or directory to place the "
+                                  "executable.".format(exec_path, install_dir))
+
+
+def main():
+    verify_python_version()
+    tmp_dir = create_tmp_dir()
+    install_dir = get_install_dir()
+    exec_dir = get_exec_dir()
+    verify_install_dir_exec_path_conflict(install_dir, exec_dir)
+    create_virtualenv(install_dir)
+    install_cli(install_dir, tmp_dir)
+    exec_filepath = create_executable(exec_dir, install_dir)
+    try:
+        handle_path_and_tab_completion(exec_filepath, exec_dir)
+    except Exception as e:
+        print_status("Unable to set up PATH. ERROR: {}".format(str(e)))
+    shutil.rmtree(tmp_dir)
+    print_status("Installation successful.")
+    print_status("Run the CLI with {} --help".format(exec_filepath))
+
+
+if __name__ == '__main__':
+    try:
+        main()
+    except CLIInstallError as cie:
+        print('ERROR: '+str(cie), file=sys.stderr)
+        sys.exit(1)
+    except KeyboardInterrupt:
+        print('\n\nExiting...')
+        sys.exit(1)
+
+# SIG # Begin signature block
```

Comment:
how do you produce the signature block?

Reply:
use sign.py to generate signaure.
and need use validate.py to validate the script if add signature in the install.py.

https://ms.portal.azure.com/#view/Microsoft_Azure_Storage/ContainerMenuBlade/~/overview/storageAccountId/%2Fsubscriptions%2F96aede12-2f73-41cb-b983-6d11a904839b%2FresourceGroups%2Fpromptflow-prod%2Fproviders%2FMicrosoft.Storage%2FstorageAccounts%2Fpromptflowartifact/path/linux-install-scripts/etag/%220x8DC0D142E2009BF%22/defaultEncryptionScope/%24account-encryption-key/denyEncryptionScopeOverride~/false/defaultId//publicAccessVal/Container